src/Handlers/SystemWeb/HttpHandlerEndpointConventionBuilder.cs

```diff
@@ -18,13 +18,19 @@ internal sealed class HttpHandlerEndpointConventionBuilder : EndpointDataSource,
     private readonly AdditionalManager _additional;
     private readonly IHttpHandlerCollection[] _managers;
     private readonly HandlerMetadataProvider _metadataProvider;
+    private readonly RequestDelegate _defaultHandler;
+
     private List<Action<EndpointBuilder>> _conventions = [];
 
-    public HttpHandlerEndpointConventionBuilder(IEnumerable<IHttpHandlerCollection> managers, HandlerMetadataProvider metadataProvider)
+    public HttpHandlerEndpointConventionBuilder(
+        IEnumerable<IHttpHandlerCollection> managers,
+        HandlerMetadataProvider metadataProvider,
+        IServiceProvider services)
     {
         _additional = new AdditionalManager();
         _managers = [.. managers, _additional];
         _metadataProvider = metadataProvider;
+        _defaultHandler = BuildDefaultHandler(services);
```

Comment:
can we keep serviceprovider as a local property for further use ? (or you don't see in future use)

Reply:
if we need it, we can change it. I'd rather not keep fields around that aren't serving a purpose.